SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/LegacyLoadBalancingStrategy.cpp

```diff
@@ -141,13 +156,76 @@ void FLegacyLoadBalancing::OnWorkersDisconnected(TArrayView<FLBWorkerHandle> Dis
 {
 	// Not handled
 }
-
-void FLegacyLoadBalancing::TickPartitions(FPartitionManager& PartitionMgr)
+void FLegacyLoadBalancing::TickPartitions()
 {
+	FPartitionManager& PartitionMgr = SharedData->PartitionManager;
+
 	if (bCreatedPartitions)
 	{
+		if (ServerWorkerCustomAssignment->GetModifiedEntities().Num() > 0 || DebugCompStorage->GetModifiedEntities().Num() > 0)
+		{
+			for (const auto& Layer : LBContext.Layers)
+			{
+				for (const auto& Cell : Layer.Cells)
+				{
+					uint32 WorkerIdx = Cell.WorkerId - 1;
+					FPartitionHandle WorkerPartition = Partitions[WorkerIdx];
+					FLBWorkerHandle WorkerHandle = VirtualWorkerIdToHandle[WorkerIdx];
```

Comment:
These arrays are built together, so the assertion built-in the array should be enough.

Reply:
Personally I'd exclude an explicit assert with associated message, perhaps outside the loops, since this will be clearer for the reader - optional/nit however, feel free to resolve conversation if you wish.